src/app/atoms/CollectibleImage.tsx

```diff
@@ -1,24 +1,35 @@
 import React, { FC, useState } from 'react';
 
-import { formatImgUri, sanitizeImgUri } from 'lib/image-uri';
-import { AssetMetadata } from 'lib/temple/metadata/types';
+import { formatCollectibleUri } from 'lib/image-uri';
+import { useRetryableSWR } from 'lib/swr';
+import { fromAssetSlug } from 'lib/temple/assets';
+import { useTezos } from 'lib/temple/front';
+import { AssetMetadata } from 'lib/temple/metadata';
 
 interface Props {
   collectibleMetadata: AssetMetadata;
   Placeholder: React.FunctionComponent<any>;
   className?: string;
+  assetSlug: string;
 }
 
-const CollectibleImage: FC<Props> = ({ collectibleMetadata, Placeholder, className }) => {
+const CollectibleImage: FC<Props> = ({ collectibleMetadata, assetSlug, Placeholder, className }) => {
+  const tezos = useTezos();
   const [isLoaded, setIsLoaded] = useState(false);
+  const asset = useRetryableSWR(['asset', assetSlug, tezos.checksum], () => fromAssetSlug(tezos, assetSlug), {
+    suspense: true
+  }).data!;
+  if (asset === 'tez') return null;
+  const assetId = asset.id ? asset.id.toString() : '0';
```

Comment:
it's better to use `toFixed` method for `BigNumber` as `toString` sometimes return wrong text